Need dependency on importlib-metadata

<!--
Thanks for reporting your issue.
Please fill out the sections below.
-->

- [X] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

Sphinx 4.4.0 introduced a dependency on the `importlib-metadata` backport when installed with python < 3.10. (See sphinx-doc/sphinx#10007.) The 4.4.0 conda package does not have that dependency metadata, so installing sphinx 4.4.0 via conda-forge does not install `importlib-metadata` when it should.


Example:

First make an empty project.
```shell
$ conda create -n test -y "sphinx=4.4.0" "python<3.10"
$ conda activate test
$ mkdir test-dir
$ cd test-dir
$ sphinx-quickstart
< accept default options, put in "test" as name, responses don't matter >
```

The way I have found to trigger an error is to use a non-default theme. So I edit the `conf.py` to set `html_theme = 'my_custom_theme'`. It doesn't actually matter what the theme is, I haven't installed it, but it will never be found because of an error.

```shell
$ make html
Running Sphinx v4.4.0
loading pickled environment... done

Exception occurred:
  File "/usr/local/Caskroom/miniconda/base/envs/test/lib/python3.9/site-packages/sphinx/theming.py", line 207, in load_external_theme
    theme_entry_points = entry_points(group='sphinx.html_themes')
TypeError: entry_points() got an unexpected keyword argument 'group'
The full traceback has been saved in /var/folders/rn/37gxvfws6mvcyw4x6k18f_8m0000gp/T/sphinx-err-xhwhv03w.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 2
```

If I then install `importlib-metadata`, I get the (correct, expected) error about my fake theme.
```shell
$ conda install -y importlib-metadata
$ make html
Running Sphinx v4.4.0
loading pickled environment... done

Theme error:
no theme named 'my_custom_theme' found (missing theme.conf?)
make: *** [html] Error 2
```

--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
 About sphinx
 ============
 
-Home: http://www.sphinx-doc.org
+Home: https://www.sphinx-doc.org
 
 Package license: BSD-2-Clause
 
@@ -11,7 +11,7 @@ Summary: Sphinx is a tool that makes it easy to create intelligent and beautiful
 
 Development: https://github.com/sphinx-doc/sphinx
 
-Documentation: http://www.sphinx-doc.org/en/stable/contents.html
+Documentation: https://www.sphinx-doc.org/en/stable/contents.html
 
 Sphinx is a tool that makes it easy to create intelligent and beautiful documentation,
 written by Georg Brandl and licensed under the BSD license.

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -29,6 +29,7 @@ requirements:
     - babel >=1.3
     - docutils >=0.14,<0.18
     - imagesize
+    - importlib-metadata >=4.4
     - jinja2 >=2.3
     - packaging
     - pygments >=2.0
@@ -56,7 +57,7 @@ test:
     - sphinx-quickstart --version
 
 about:
-  home: http://www.sphinx-doc.org
+  home: https://www.sphinx-doc.org
   license: BSD-2-Clause
   license_file: LICENSE
   summary: Sphinx is a tool that makes it easy to create intelligent and beautiful documentation
@@ -66,7 +67,7 @@ about:
     It was originally created for the new Python documentation, and it has excellent
     facilities for the documentation of Python projects, but C/C++ is already supported
     as well, and it is planned to add special support for other languages as well.
-  doc_url: http://www.sphinx-doc.org/en/stable/contents.html
+  doc_url: https://www.sphinx-doc.org/en/stable/contents.html
   dev_url: https://github.com/sphinx-doc/sphinx
 
 extra:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -9,7 +9,7 @@ source:
   sha256: 6caad9786055cb1fa22b4a365c1775816b876f91966481765d7d50e9f0dd35cc
 
 build:
-  number: 0
+  number: 1
   noarch: python
   script: {{ PYTHON }} -m pip install . -vv
   entry_points:
@@ -35,7 +35,6 @@ requirements:
     - pygments >=2.0
     - python
     - requests >=2.5.0
-    - setuptools
     - snowballstemmer >=1.1
     - sphinxcontrib-applehelp
     - sphinxcontrib-devhelp

